src/api.rs

```diff
@@ -13,10 +13,69 @@ pub struct Features {
     pub features: Vec<Feature>,
 }
 
+#[derive(Debug, PartialEq, Clone)]
+pub struct TagFilter {
+    pub name: String,
+    pub value: String,
+}
+
+impl TagFilter {
+    pub fn format(&self) -> String {
+        format!("{}:{}", self.name, self.value)
+    }
+}
+
+#[derive(Serialize, Debug)]
+struct FeaturesQuery {
+    pub project: Option<String>,
+    #[serde(rename = "namePrefix")]
+    pub name_prefix: Option<String>,
+    #[serde(rename = "tag")]
+    pub tags: Option<Vec<String>>,
+}
+
 impl Features {
     pub fn endpoint(api_url: &str) -> String {
         format!("{}/client/features", api_url)
     }
+
+    #[cfg(feature = "surf")]
+    pub fn query(
+        project: Option<String>,
+        name_prefix: Option<String>,
+        tags: &Option<Vec<TagFilter>>,
+    ) -> impl Serialize + std::fmt::Debug {
+        FeaturesQuery {
+            project,
+            name_prefix,
+            tags: tags.as_ref().map(|tags| tags.iter().map(|tag| tag.format()).collect()),
+        }
+    }
+
+    #[cfg(not(feature = "surf"))]
```

Comment:
these parts should be written against the local http client trait, so that both surf and reqwest users gain them. That trait may need tweaking to accomodate your needs.

Reply:
I couldn't find a "nice" way of implementing it :thinking: I didn't want to polute the trait with the logic of features query, so instead I made it that the query is already present in the url when it's passed to the http layer. wdyt is it fine to do it that way?